sky/tools/create_macos_framework.py

```diff
@@ -93,6 +94,16 @@ def main():
   ])
   process_framework(dst, args, fat_framework, fat_framework_binary)
 
+  # Set fat_framework file attributes to read and execute for all.
+  subprocess.check_call([
```

Comment:
It's not really great to make non-script/binary files executable. It could be a security hole if one of them starts with `#!`, for instance

Ideally, you'd take whatever permissions the file already has for owner and give it the same permissions for other and group.

I'd suggest replacing these two calls with something like this:

```python
# Add group and other readability to all files.
subprocess.check_call([
  'chmod', '-R', 'og+r',
  os.path.join(fat_framework, 'Versions')
])
# Find all the files below the target dir with owner execute permission
find_subprocess = subprocess.Popen(["find", os.path.join(fat_framework, 'Versions', 'A', 'FlutterMacOS'), "-type", "f", "-perm", "-100", "-print0"], stdout=subprocess.PIPE)
# Add execute permission for other and group for all files that had it for owner.
xargs_subprocess = subprocess.Popen(["xargs", "-0", "chmod", "og+x"], stdin=find_subprocess.stdout)
find_subprocess.wait()
xargs_subprocess.wait()
```

You might also consider using the "Versions" path in the `find` subprocess instead of just the `FlutterMacOS` subdir, to make sure all the paths get the same treatment.

Reply:
done